crates/octez/src/node_config.rs

```diff
@@ -0,0 +1,278 @@
+use anyhow::Result;
+use std::path::PathBuf;
+
+pub const DEFAULT_RPC_ENDPOINT: &str = "localhost:8732";
+const DEFAULT_NETWORK: &str = "sandbox";
+const DEFAULT_BINARY_PATH: &str = "octez-node";
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum OctezNodeHistoryMode {
+    Archive,
+    // The numerical value represents additional cycles preserved. 0 is acceptable.
+    Full(u8),
+    Rolling(u8),
+}
+
+impl ToString for OctezNodeHistoryMode {
+    fn to_string(&self) -> String {
+        match self {
+            Self::Archive => "archive".to_owned(),
+            Self::Full(v) => format!("full:{v}"),
+            Self::Rolling(v) => format!("rolling:{v}"),
+        }
+    }
+}
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct OctezNodeRunOptions {
+    synchonisation_threshold: u8,
+    network: String,
+    history_mode: Option<OctezNodeHistoryMode>,
+}
+
+impl ToString for OctezNodeRunOptions {
+    fn to_string(&self) -> String {
+        let mut s = vec![];
+        s.push(format!(
+            "--synchronisation-threshold {}",
+            &self.synchonisation_threshold
+        ));
+        s.push(format!("--network {}", &self.network));
+        if let Some(v) = &self.history_mode {
+            s.push(format!("--history-mode {}", v.to_string()));
+        }
+        s.join(" ")
+    }
+}
+
+#[derive(Default)]
+pub struct OctezNodeRunOptionsBuilder {
+    synchonisation_threshold: Option<u8>,
+    network: Option<String>,
+    history_mode: Option<OctezNodeHistoryMode>,
+}
+
+impl OctezNodeRunOptionsBuilder {
+    pub fn new() -> Self {
+        OctezNodeRunOptionsBuilder {
+            ..Default::default()
+        }
+    }
+
+    pub fn set_synchronisation_threshold(&mut self, threshold: u8) -> &mut Self {
+        self.synchonisation_threshold.replace(threshold);
+        self
+    }
+
+    pub fn set_network(&mut self, network: &str) -> &mut Self {
+        self.network.replace(network.to_owned());
+        self
+    }
+
+    pub fn set_history_mode(&mut self, mode: OctezNodeHistoryMode) -> &mut Self {
+        self.history_mode.replace(mode);
+        self
+    }
+
+    pub fn build(&mut self) -> OctezNodeRunOptions {
+        OctezNodeRunOptions {
+            synchonisation_threshold: self
+                .synchonisation_threshold
+                .take()
+                .unwrap_or_default(),
+            network: self.network.take().unwrap_or(DEFAULT_NETWORK.to_owned()),
+            history_mode: self.history_mode.take(),
+        }
+    }
+}
+
+#[derive(Default, Clone)]
+pub struct OctezNodeConfig {
+    /// Path to the octez node binary.
+    pub binary_path: PathBuf,
+    /// Path to the directory where the node keeps data.
+    pub data_dir: PathBuf,
+    /// Name of the tezos network that the node instance runs on.
+    pub network: String,
+    /// HTTP endpoint of the node RPC interface, e.g. 'localhost:8732'
+    pub rpc_endpoint: String,
```

Comment:
both the client and baker uses the `Endpoint` struct, probabbly better to use that?

Reply:
I agree, but I would keep it as is for now and do this in a separate PR. Changing this requires moving the endpoint struct to octez crate and updating the octez node implementation, which are beyond the scope of this PR.